controllers/complianceeventsapi/server.go

```diff
@@ -813,6 +828,201 @@ func postComplianceEvent(db *sql.DB, w http.ResponseWriter, r *http.Request) {
+		writeErrMsgJSON(w, "Internal Error", http.StatusInternalServerError)
+
+		return
+	}
+
+	defer rows.Close()
+
+	headers := getCsvHeader(queryArgs.IncludeSpec)
+
+	writer := csv.NewWriter(w)
+
+	err = writer.Write(headers)
+	if err != nil {
+		log.Error(err, "Failed to write csv header")
+		writeErrMsgJSON(w, "Internal Error", http.StatusInternalServerError)
+
+		return
+	}
+
+	for rows.Next() {
+		ce, err := scanIntoComplianceEvent(rows, queryArgs.IncludeSpec)
+		if err != nil {
+			log.Error(err, "Failed to unmarshal the database results")
+			writeErrMsgJSON(w, "Internal Error", http.StatusInternalServerError)
+
+			return
+		}
+
+		stringValues := convertToCsvLine(ce, queryArgs.IncludeSpec)
+
+		err = writer.Write(stringValues)
+		if err != nil {
+			log.Error(err, "Failed to write csv list")
+			writeErrMsgJSON(w, "Internal Error", http.StatusInternalServerError)
+
+			return
+		}
+	}
+
+	w.Header().Set("Content-Disposition", "attachment; filename=reports.csv")
+	w.Header().Set("Content-Type", "text/csv")
+	// It's going to be divided into chunks. if the user don't get it all at once,
+	// the user can receive one by one in the meantime
+	w.Header().Set("Transfer-Encoding", "chunked")
+
+	writer.Flush()
+}
+
+func convertToCsvLine(ce *ComplianceEvent, includeSpec bool) []string {
+	nilString := ""
+
+	if ce.ParentPolicy == nil {
+		ce.ParentPolicy = &ParentPolicy{
+			KeyID:      0,
+			Name:       "",
+			Namespace:  "",
+			Categories: nil,
+			Controls:   nil,
+			Standards:  nil,
+		}
+	}
+
+	if ce.Event.ReportedBy == nil {
+		ce.Event.ReportedBy = &nilString
+	}
+
+	if ce.Policy.Severity == nil {
+		ce.Policy.Severity = &nilString
+	}
+
+	if ce.Policy.Namespace == nil {
+		ce.Policy.Namespace = &nilString
+	}
+
+	values := []string{
+		convertToString(ce.EventID),
+		convertToString(ce.Event.Compliance),
+		convertToString(ce.Event.Message),
+		convertToString(ce.Event.Metadata),
+		convertToString(*ce.Event.ReportedBy),
+		convertToString(ce.Event.Timestamp),
+		convertToString(ce.Cluster.ClusterID),
+		convertToString(ce.Cluster.Name),
+		convertToString(ce.ParentPolicy.KeyID),
+		convertToString(ce.ParentPolicy.Name),
+		convertToString(ce.ParentPolicy.Namespace),
+		convertToString(ce.ParentPolicy.Categories),
+		convertToString(ce.ParentPolicy.Controls),
+		convertToString(ce.ParentPolicy.Standards),
+		convertToString(ce.Policy.KeyID),
+		convertToString(ce.Policy.APIGroup),
+		convertToString(ce.Policy.Kind),
+		convertToString(ce.Policy.Name),
+		convertToString(*ce.Policy.Namespace),
+		convertToString(*ce.Policy.Severity),
+	}
+
+	if includeSpec {
+		values = append(values, convertToString(ce.Policy.Spec))
+	}
+
+	return values
+}
+
+func convertToString(v interface{}) string {
+	switch vv := v.(type) {
+	case *string:
+		if vv == nil {
+			return ""
+		}
+
+		return *vv
+	case string:
+		return vv
+	case int32:
+		// All int32 related id
+		if int(vv) == 0 {
+			return "nil"
```

Comment:
I think no value is more readable in a spreadsheet.
```suggestion
			return ""
```

Reply:
How about 
```
 	case JSONMap:
		if vv == nil {
			return "nil"
		}
```
Since most nil cases change to ""

Reply:
Yeah, I think so.

Reply:
jsonByte, err := json.MarshalIndent(vv, "", "  ") output like this:
json```
"{
  ""test"": true
}"
```

Reply:
But in Mac spreadsheet and slack spreadsheet looks fine 
```
{
  "severity": "low",
  "test": "one"
}
```